airbyte-webapp/src/components/base/Button/types.tsx

```diff
@@ -1,10 +1,22 @@
+import React from "react";
+
+export enum ButtonType {
+  Primary = "primary",
+  Secondary = "secondary",
+  Danger = "danger",
+  LightGrey = "lightGrey",
+}
+
 export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  customStyles?: string;
+  buttonType?: ButtonType;
+  clickable?: boolean;
   full?: boolean;
-  danger?: boolean;
-  secondary?: boolean;
+  icon?: React.ReactElement;
+  iconPosition?: "left" | "right";
   isLoading?: boolean;
-  iconOnly?: boolean;
+  label?: string | React.ReactNode;
```

Comment:
`string` is part of `ReactNode`.

Also, I do like it better when this is the `children`, which is most common with other design libraries.